lib/install/management/store_files.go

```diff
@@ -201,20 +214,24 @@ func (d *Dispatcher) getChildren(ds *object.Datastore, dsPath string) ([]string,
 }
 
 // getSortedChildren returns all children under datastore path in reversed order.
-func (d *Dispatcher) getSortedChildren(ds *object.Datastore, dsPath string) ([]string, error) {
-	result, err := d.getChildren(ds, dsPath)
+func (d *Dispatcher) getSortedChildren(ds *object.Datastore, dsPath string, matchPattern ...string) ([]string, error) {
+	result, err := d.getChildren(ds, dsPath, matchPattern...)
 	if err != nil {
 		return nil, err
 	}
 	sort.Sort(sort.Reverse(sort.StringSlice(result)))
 	return result, nil
 }
 
-func (d *Dispatcher) lsSubFolder(ds *object.Datastore, dsPath string) (*types.ArrayOfHostDatastoreBrowserSearchResults, error) {
+func (d *Dispatcher) lsSubFolder(ds *object.Datastore, dsPath string, matchPattern ...string) (*types.ArrayOfHostDatastoreBrowserSearchResults, error) {
 	defer trace.End(trace.Begin(dsPath))
 
+	// match every subfolers by default
+	if len(matchPattern) == 0 {
+		matchPattern = append(matchPattern, "*")
+	}
 	spec := types.HostDatastoreBrowserSearchSpec{
-		MatchPattern: []string{"*"},
+		MatchPattern: matchPattern,
```

Comment:
@emlin 
1. deleteIteratively -> getSortedChildren -> getChildren which only collects files. so tests1/abc is never returned;
2. But you're right, if there were other nested directories, they will not be cleaned up;
3. Because in the original implementation, their removal is achieved indirectly by removing the root directory